packages/core/src/types.ts

```diff
@@ -309,11 +688,16 @@ interface InlineConfigElement {
 /**
  * Represents the basic interface for a source code object.
  */
-interface SourceCodeBase {
+interface SourceCodeBase<
+	LangOptions = LanguageOptions,
```

Comment:
[Question] Can `LangOptions` be something that isn't assignable to `LanguageOptions`? Put another way: is it intentional that there's no `extends LanguageOptions` here?

Reply:
`LanguageOptions` itself doesn't have any keys by default, so it seemed like `extends LanguageOptions` wasn't necessary? 

I was actually trying to reason through this choice, too, so any insights appreciated.